### PR TITLE
Unit test for Input component

### PR DESCRIPTION
**Pull request by namnguyen20999**
| Test Coverage Progress | File            | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s |
|------------------------|-----------------|---------|----------|---------|---------|-------------------|
| :hourglass: Before     | Input.tsx            |   89.28 |    72.81 |      68 |   92.45 | 35-38,89-90,184,187,213 
| :white_check_mark: After  | Input.tsx        |     100 |    85.43 |     100 |     100 | 110-111,119-120,128,159-160,178-179,269,289

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
🙂

File: frontend/taipy-gui/src/components/Taipy/Input.tsx [hunk 1]
```diff
@@ -235,14 +235,14 @@ const Input = (props: TaipyInputProps) => {
                       endAdornment: (
                           <div style={verticalDivStyle}>
                               <IconButton
-                                  data-testid="stepper-up-spinner"
+                                  aria-label={"Increment " + id}
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Input.tsx]: Aria-label is used by automatic readers to help people with disabilities. 
We need to give it a value that is meaningful
I propose `increment value` (no id)

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Well done

File: frontend/taipy-gui/src/components/Taipy/Input.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Input.tsx]: updated

**Review by FabienLelaquais — DISMISSED** [review]
👌

File: frontend/taipy-gui/src/components/Taipy/Input.spec.tsx [hunk 2]
```diff
@@ -128,6 +147,18 @@ describe("Input Component", () => {
...
     });
+    it("should display visibility off icon when showPassword is true", async () => {
+        const { getByLabelText } = render(<Input value={"Test Input"} type="password" />);
+        const visibilityButton = getByLabelText("toggle password visibility");
+        fireEvent.click(visibilityButton);
+        const visibilityIcon = document.querySelector('svg[data-testid="VisibilityOffIcon"]');
+        expect(visibilityIcon).toBeInTheDocument();
+    });
+    it("should display visibility icon when showPassword is false", async () => {
+        const { getByLabelText } = render(<Input value={"Test Input"} type="password" />);
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Input.spec.tsx:158]: not testing the icon's presence ?

File: frontend/taipy-gui/src/components/Taipy/Input.tsx [hunk 3]
```diff
@@ -235,14 +235,14 @@ const Input = (props: TaipyInputProps) => {
                       endAdornment: (
                           <div style={verticalDivStyle}>
                               <IconButton
-                                  data-testid="stepper-up-spinner"
+                                  aria-label={"Increment value"}
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Input.tsx]: no need for curly braces {

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: frontend/taipy-gui/src/components/Taipy/Input.spec.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Input.spec.tsx:158]: it should be in this case: `should display visibility off icon when showPassword is true`. That case lay right above this case actually

File: frontend/taipy-gui/src/components/Taipy/Input.tsx (hunk 3 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Input.tsx]: updated

File: frontend/taipy-gui/src/components/Taipy/Input.spec.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Input.spec.tsx:158]: then rename the test ("should display visibility icon when showPassword is false" is misleading)

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: frontend/taipy-gui/src/components/Taipy/Input.spec.tsx (hunk 2 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Input.spec.tsx:158]: updated

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**